src/qmeta/schema/migrate-3-to-4.sql

```diff
@@ -8,3 +8,6 @@ ALTER TABLE `QInfo`
     MODIFY COLUMN `qTemplate`   MEDIUMTEXT NOT NULL     COMMENT 'Query template, string used to build final per-chunk query.',
     MODIFY COLUMN `qMerge`      MEDIUMTEXT DEFAULT NULL COMMENT 'Merge (or aggregate) query to be executed on results table, result of this query is stored in merge table. If NULL then it is equivalent to SELECT *.',
     MODIFY COLUMN `resultQuery` MEDIUMTEXT DEFAULT NULL COMMENT 'Query to be used by mysqlproxy to get final results.';
+
+-- Update record for schema version, migration script expects this record to exist.
+UPDATE `QMetadata` SET `value` = '4' WHERE `metakey` = 'version';
```

Comment:
I think this should not be done here. the SchemaMigMgr subclass applies the migration files, and is allowed to do other work (which is why the version should not be set), and setting the version should be the very last thing it does.

Reply:
(same comment applies other `migrate-` files where inserting the version has been added.

Reply:
Well, I myself was not certain if putting the version number into the SQL files was the right thing to do. I just see this in some, and not in other files. So, if you're sure this shouldn't be done in the `migrate-<from>-<to>.sql` files then perhaps we should do it consistently on all files?
@andy-slac What do you think about this?

Reply:
I think removing it from migrate- files is correct. I may have added it in some cases before I had learned more/developed an opinion. But I'm also curious what Andy will say. (Probably something I haven't thought of entirely haha)

Reply:
I've inspected the schema migration code. and it looks like the version management is done by the package-level `schema_management.py` files, not by the main `smig` tool. I've cleaned up the schema version update from the SQL files except for those cases where the initial version is inserted (SQL's `INSERT INTO QMetadata ...`) right after creating that table.